### ISSUE TITLE
Email images broken: "remote server's identity certificate"

### ISSUE THREAD

**Issue by stationaery** [techworkersco/techworkersco.github.io#279]
Error message in today's email via MailChimp: "image did not load because remote server's identity certificate is invalid"

**Comment by stationaery** [comment]
May have something to do with @shushugah modifying the domains recently

**Comment by jessesquires** [comment]
Hm weird... no idea. I'll defer to @shushugah

**Comment by shushugah** [comment]
How does the mailchimp setup work? What URL's is it pointing to with the newsletter/expecting?

**Comment by jessesquires** [comment]
Mailchimp just reads the RSS feed. 

(Images use absolute URLs so it can grab them correctly.)

**Comment by shushugah** [comment]
is there any email alias used? like newsletter@techworkerscoalition.org or something? I didn't touch techworkersco.org, but techworkerscoalition.org is now on Netlify DNS.

**Comment by shushugah** [comment]
I'm not seeing absolute urls for all images tho https://news.techworkerscoalition.org/feed.xml

**Comment by stationaery** [comment]
We've got another issue coming out tomorrow. Hopefully you @jessesquires @shushugah can fix the image coming through in MailChimp before then? There were some changes discussed and made a few weeks ago - here https://github.com/techworkersco/techworkersco.github.io/pull/262#issuecomment-947386411 and here https://github.com/techworkersco/techworkersco.github.io/pull/265

**Comment by jessesquires** [comment]
@shushugah 

> is there any email alias used? like newsletter@techworkerscoalition.org or something? I didn't touch techworkersco.org, but techworkerscoalition.org is now on Netlify DNS.

Looks like emails are currently sent from `news-noreply@techworkerscoalition.org` -- which I don't remember setting up. We had originally setup a protonmail account. @stationaery Do you remember the details behind the email setup?

So maybe this is part of the issue?

> I'm not seeing absolute urls for all images tho

oh oops. 😳 well, it looks like #265 is the problem. we made the adjustments to img urls for "development" builds so that Netlify previews would work as expected

**Comment by stationaery** [comment]
> Looks like emails are currently sent from news-noreply@techworkerscoalition.org -- which I don't remember setting up. We had originally setup a protonmail account. @stationaery Do you remember the details behind the email setup?

I believe that setup was @anaulin and/or @dellsystem 😊😊

**Comment by jessesquires** [comment]
RE: img URLs and #265 

The GitHub pages environment **should** be setting `jekyll.environment` to `"production"`, but it looks like that isn't happening for some reason...

https://stackoverflow.com/questions/34603799/setting-jekyll-environment-for-github-pages

I'll investigate a bit more, and get this fixed by EOD -- at least a patch to make it work (which, might break the images in the previews. cc @stationaery)

**Comment by jessesquires** [comment]
Actually -- it looks like we published multiple issues successfully **after** #265. 

Only the most recent one has this issue. So... not sure what has changed...

**Comment by stationaery** [comment]
The only thing that changed recently was Yonatan tweaking the domains

**Comment by shushugah** [comment]
I do not see any TXT or MX records for the email, but that can be changed if one of you who has access to protonmail goes through the steps and sends me a unique **txt record** (signal works best; my number: +4915166008114) but also my twitter dm is public https://twitter.com/shushugah per the instructions here: https://protonmail.com/support/knowledge-base/dns-records/

Do you know if DMARK/SPF/DKIM anti spoofing was set up? If it isn't, the chances of such emails ending up in spam is quite high unfortunately.

**Comment by anaulin** [comment]
Re: email, I didn't set any of that up, and don't have any context.

**Comment by shushugah** [comment]
@jessesquires I tagged you in #282 am unsure what the linter error is, but let me know if it makes sense. That would address the broken feed.xml

**Comment by jessesquires** [comment]
@shushugah #282 looks good! thank you for that! 🙌🏼 

> I do not see any TXT or MX records for the email, but that can be changed if one of you who has access to protonmail goes through the steps and sends me a unique txt record 

Oh interesting. I think @kidmillions set up the email maybe? But I doubt we'll hear from him. 😅 

> Do you know if DMARK/SPF/DKIM anti spoofing was set up? If it isn't, the chances of such emails ending up in spam is quite high unfortunately.

Possibly, but I'm not sure.

@stationaery Could you sign-in to mailchimp and confirm the "from" email?

**Comment by jessesquires** [comment]
Returning to the original error message from @stationaery:

> Error message in today's email via MailChimp: "image did not load because remote server's identity certificate is invalid"

I think the idea that something is wrong with the sender email account was a red herring. I think all that setup is actually fine.

Because hosting/serving was moved from GH-Pages to Netlify (#282), we were accidentally using relative image urls (e.g. `/img/vol3-header.jpg`) instead of absolute urls (e.g., `news.techworkersco.org/img/vol3-header.jpg`) (#265). 

Then MailChimp was trying to resolve the relative url `/img/...` against `mailchimp.com` or just some other garbage, and it just has a shitty/misleading error message.

In other words: I think we can close this as fixed now because of #282.

**Comment by stationaery** [comment]
> Do you know if DMARK/SPF/DKIM anti spoofing was set up?

I don't but @dellsystem perhaps you know?

**Comment by jessesquires** [comment]
Thanks @stationaery ! 🙌🏼 

Ok -- I think we're good here and I'm going to close this. The emails went out this morning for #280 and everything looked good!

I'm pretty sure we setup all DNS records correctly way back when the `noreply` email was created.
